shared/src/commonMain/kotlin/kmp/shared/extension/Store.kt

```diff
@@ -0,0 +1,70 @@
+package kmp.shared.extension
+
+import kmp.shared.base.ErrorResult
+import kmp.shared.base.ErrorResultException
+import kmp.shared.base.Result
+import kmp.shared.base.error.domain.CommonError
+import kmp.shared.infrastructure.model.UserDto
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.filterNot
+import kotlinx.coroutines.flow.first
+import kotlinx.coroutines.flow.map
+import org.mobilenativefoundation.store.store5.MutableStore
+import org.mobilenativefoundation.store.store5.StoreReadRequest
+import org.mobilenativefoundation.store.store5.StoreReadResponse
+import org.mobilenativefoundation.store.store5.StoreWriteResponse
+
+fun <Key : Any, Output : Any> MutableStore<Key, Output>.getResultFlow(key: Key): Flow<Result<Output>> =
+    stream<UserDto>(StoreReadRequest.cached(key, refresh = true))
```

Comment:
`UserDto` should probably not be here right?

Reply:
yep, thanks ❤️